src/TempleLineOfCredit.sol

```diff
@@ -98,125 +127,177 @@ contract TempleLineOfCredit is Ownable, Operators {
+        }
+    }
+
     /**
      * @dev Allows operator to depoist debt tokens
      * @param amount is the amount to deposit
      */
-    function depositDebt(address account, uint256 amount) external onlyOperators{
+    function depositDebt(address debtToken, address account, uint256 amount) external onlyOperators{
         if (amount == 0) {
             revert InvalidAmount(amount);
         }
-        debtBalance += amount;
-        debtToken.safeTransferFrom(
+        if (!debtTokens[debtToken].isAllowed || debtTokens[debtToken].tokenType != TokenType.TRANSFER ) revert Unsupported(debtToken);
+        IERC20(debtToken).safeTransferFrom(
             account,
             address(this),
             amount
         );
-        emit DepositDebt(amount);
+        emit DepositDebt(debtToken, amount);
     }
 
         /**
      * @dev Allows operator to remove debt token
      * @param amount is the amount to remove
      */
-    function removeDebt(uint256 amount) external onlyOperators{
+    function removeDebt(address debtToken, address account, uint256 amount) external onlyOperators{
         if (amount == 0) {
             revert InvalidAmount(amount);
         }
-        debtBalance -= amount;
-        debtToken.safeTransfer(
+        if (!debtTokens[debtToken].isAllowed || debtTokens[debtToken].tokenType != TokenType.TRANSFER ) revert Unsupported(debtToken);
+        IERC20(debtToken).safeTransfer(
             msg.sender,
             amount
         );
-        emit RemoveDebt(amount);
+        emit RemoveDebt(debtToken, amount);
     }
 
     /**
      * @dev Allows borrower to deposit collateral
-     * @param amount is the amount to deposit
+     * @param collateralAmount is the amount to deposit
      */
-    function postCollateral(uint256 amount) external {
-        if (amount == 0) revert InvalidAmount(amount);
-        positions[msg.sender].collateralAmount += amount;
+    function postCollateral(uint256 collateralAmount) external {
+        if (collateralAmount == 0) revert InvalidAmount(collateralAmount);
+        positions[msg.sender].collateralAmount += collateralAmount;
         collateralToken.safeTransferFrom(
             msg.sender,
             address(this),
-            amount
+            collateralAmount 
         );
-        emit PostCollateral(msg.sender, amount);
+        emit PostCollateral(msg.sender, collateralAmount);
     }
 
-    function borrow(uint256 borrowAmount) external {
-        if (borrowAmount == 0) revert InvalidAmount(borrowAmount);
+    function borrow(address[] memory tokens, uint256[] memory borrowAmounts) external {
 
-        uint256 debtAmount =  positions[msg.sender].debtAmount;
-        if (borrowAmount > debtBalance) {
-            revert InsufficentDebtToken(debtBalance, borrowAmount);
-        }
-        if (debtAmount != 0) {
-            debtAmount = getTotalDebtAmount(msg.sender);
-        }
+        for (uint256 i =0; i < tokens.length; i++) {
```

Comment:
A few minor gas savings with:

1.
```
for (uint256 i; i < tokens.length; ++i) {
```
Same for other loops below too.

2. Move the declaration of vars above the loop, ie:
```
address debtToken;
uint256 borrowAmount;
DebtToken memory debtTokenInfo;
...
for (uint256 i; i < tokens.length; ++i) {
    ...
}
```
So they don't have to be re-allocated each iteration.


Should also check that `tokens.length == borrowAmounts.length`

Reply:
👍